oidc/transport.go

```diff
@@ -47,24 +47,24 @@ type AuthenticatedTransport struct {
 	http.RoundTripper
 
 	mu  sync.Mutex
```

Comment:
Ah. I think your solution is more complicated. It's also less generally useful - my way would work for an AuthenticatedTransport with any TokenRefresher